newIDE/app/src/EventsSheet/ParameterFields/ObjectVariableField.js

```diff
@@ -45,17 +45,49 @@ export default React.forwardRef<ParameterFieldProps, ParameterFieldInterface>(
     });
 
     const { layout } = scope;
-    let object = null;
-    let variablesContainer = null;
-    if (objectName) {
-      if (layout && layout.hasObjectNamed(objectName)) {
-        object = layout.getObject(objectName);
-        variablesContainer = object.getVariables();
-      } else if (project && project.hasObjectNamed(objectName)) {
-        object = project.getObject(objectName);
-        variablesContainer = object.getVariables();
-      }
-    }
+    const object = objectName
+      ? layout && layout.hasObjectNamed(objectName)
+        ? layout.getObject(objectName)
+        : project && project.hasObjectNamed(objectName)
+        ? project.getObject(objectName)
+        : null
+      : null;
+    const variablesContainers = React.useMemo<Array<gdVariablesContainer>>(
+      () => {
+        if (!objectName) {
+          return [];
+        }
+        const variablesContainers: Array<gdVariablesContainer> = [];
+        if (object) {
+          variablesContainers.push(object.getVariables());
+        } else {
+          const group =
+            layout && layout.getObjectGroups().has(objectName)
+              ? layout.getObjectGroups().get(objectName)
+              : project && project.getObjectGroups().has(objectName)
+              ? project.getObjectGroups().get(objectName)
+              : null;
```

Comment:
Same idea here: 

```suggestion
          const group = getObjectGroupByName(project, layout, objectName);
```

And like the other, this is a "dumb" helper but that will progressively be converted to usage of gdObjectsContainersList.